docs/cookbook.rst

```diff
@@ -0,0 +1,3 @@
+=================
+Rasterio Cookbook
```

Comment:
@perrygeo should this have more to at, at least a `.. todo::` and some brief ideas, or should this file be left out at this point if it is empty?